pkg/sql/logictest/testdata/logic_test/alter_primary_key

```diff
@@ -578,7 +578,7 @@ t  CREATE TABLE public.t (
 statement ok
 CREATE INDEX i ON t (x);
 
-statement error pq: constraint with name i already exists
+statement error pgcode 42P07 an index named "i" already exists or is being dropped
```

Comment:
i wonder if we should keep "constraint" here instead of "index" in the error message. seems more postgres-like

Reply:
My rationale for this was that since I'm returning `pgcode.DuplicateRelation` I might as well reference a relation. I don't know what the right thing to do is here and I'm not familiar with postgres. Should I change the message to s/constraint/index only, or should I also change the pgcode to something else? I'll defer to you.

Reply:
Hmm having looked into this further, it turns out that we sometimes return `pgcode.DuplicateObject` and a `duplicate constraint name` message, depending on which statements are executed.

For the sake of uniformity perhaps we should always return `pgcode.DuplicateObject` and not talk about indexes.